core/core-load-paths.el

```diff
@@ -20,83 +20,121 @@
+  (concat spacemacs-banner-directory "img/gplv3.png")
   "GPLv3 official badge image.")
-(defconst spacemacs-cache-directory
-  (expand-file-name (concat user-emacs-directory ".cache/"))
-  "Spacemacs storage area for persistent files")
-(defconst spacemacs-auto-save-directory
-  (expand-file-name (concat spacemacs-cache-directory "auto-save/"))
-  "Spacemacs auto-save directory")
+
+;; ~/.emacs.d/core/info
+(defconst spacemacs-info-directory
+  (concat spacemacs-core-directory "info/")
+  "Spacemacs info files directory.")
+
+;; ~/.emacs.d/core/info/release-notes
+(defconst spacemacs-release-notes-directory
+  (concat spacemacs-info-directory "release-notes/")
+  "Spacemacs release notes directory.")
+
+;; ~/.emacs.d/doc
 (defconst spacemacs-docs-directory
-  (expand-file-name (concat spacemacs-start-directory "doc/"))
+  (concat spacemacs-start-directory "doc/")
   "Spacemacs documentation directory.")
+
+;; ~/.emacs.d/news
 (defconst spacemacs-news-directory
-  (expand-file-name (concat spacemacs-start-directory "news/"))
+  (concat spacemacs-start-directory "news/")
   "Spacemacs News directory.")
-(defconst spacemacs-assets-directory
-  (expand-file-name (concat spacemacs-start-directory "assets/"))
-  "Spacemacs assets directory.")
+
+;; ~/.emacs.d/private
+(defconst spacemacs-private-directory
+  (concat spacemacs-start-directory "private/")
+  "Spacemacs private directory.")
+
+;; ~/.emacs.d/tests
 (defconst spacemacs-test-directory
-  (expand-file-name (concat spacemacs-start-directory "tests/"))
+  (concat spacemacs-start-directory "tests/")
   "Spacemacs tests directory.")
 
-(defconst user-home-directory
-  (expand-file-name "~/")
-  "User home directory (~/).")
-(defconst pcache-directory
-  (concat spacemacs-cache-directory "pcache/"))
+;; ~/.emacs.d/.cache
+(defconst spacemacs-cache-directory
+  (concat user-emacs-directory ".cache/")
+  "Spacemacs storage area for persistent files.")
+
+;; ~/.emacs.d/.cache/auto-save
+(defconst spacemacs-auto-save-directory
+  (concat spacemacs-cache-directory "auto-save/")
+  "Spacemacs auto-save directory.")
+
+
+;;;; Setup cache directories
+
+;; TODO: Should also catch any IO error such as permission error (Apr 25 2021 Lucius)
 (unless (file-exists-p spacemacs-cache-directory)
-    (make-directory spacemacs-cache-directory))
-
-;; load paths
-(mapc 'add-to-load-path
-      `(
-        ,spacemacs-core-directory
-        ,(concat spacemacs-core-directory "libs/")
-        ,(concat spacemacs-core-directory "libs/spacemacs-theme/")
-        ,(concat spacemacs-core-directory "libs/forks/")
-        ;; ,(concat spacemacs-core-directory "aprilfool/")
-        ))
-
-;; themes
+  (make-directory spacemacs-cache-directory))
+
+(setq pcache-directory (concat spacemacs-cache-directory "pcache/"))
+
+;;;; Load Paths
+
+(defun add-to-load-path (dir)
+  "Prepend DIR to `load-path'."
+  (add-to-list 'load-path dir))
+
+;; FIXME: unused function (Apr 25 2021 Lucius)
+(defun add-to-load-path-if-exists (dir)
```

Comment:
Maybe we should mark unused functions as obsolete.  Or even better. Since this function is public Spacemacs can warn(on call) that we are going to remove it soon and that the end user should copy pasta it to their configs or tell us why we should keep it (also this warning should be disableble(is this even a word) via .spacemacs var) How about that :thinking:

Reply:
Private unused functions can be safely removed since we don't guarantee anything about them.

Reply:
I don't plan to remove unused functions in this PR. Just mark them.

Because this can be usable in other parts.

Reply:
Sure but marking them via code instead of comment will allow us to know when/if this function is used.

Reply:
or actually anywhere :thinking:

Reply:
User normally should not use spacemacs functions anyway...

No guarantee of API stability for <1.0 release.

So I don't bother to add any error warning.

Reply:
> No guarantee of API stability for <1.0 release.

Oh boi.